profiling/src/profile/mod.rs

```diff
@@ -561,6 +679,50 @@ impl Profile {
             .get(&local_root_span_id)
             .map(i64::clone))
     }
+
+    fn upscale_values(&self, values: &[i64], labels: &[Label]) -> anyhow::Result<Vec<i64>> {
+        let mut new_values = values.to_vec();
+
+        if !self.upscaling_rules.is_empty() {
+            let mut values_to_update: Vec<usize> = vec![0; self.sample_types.len()];
+
+            // get bylabel rules first (if any)
+            let mut group_of_rules = labels
+                .iter()
+                .filter_map(|label| self.upscaling_rules.get(&(label.key, label.str)))
+                .collect::<Vec<&Vec<UpscalingRule>>>();
+
+            // get byvalue rules if any
+            if let Some(byvalue_rules) = self.upscaling_rules.get(&(0, 0)) {
+                group_of_rules.push(byvalue_rules);
+            }
+
+            // check for collision(s)
+            group_of_rules.iter().for_each(|rules| {
+                rules.iter().for_each(|rule| {
+                    rule.values_offset
+                        .iter()
+                        .for_each(|offset| values_to_update[*offset] += 1)
+                })
+            });
+
+            anyhow::ensure!(
+                values_to_update.iter().all(|v| *v < 2),
+                "Multiple rules modifying the same offset for this sample"
+            );
```

Comment:
It depends what we want: partial upscaling or no upscaling in case we have rules collision.
As you might read, I went for `do not touch my values if you intend to fail`

Reply:
Side because I tried: you cannot return a value from a `for_each`. Type incompatibility (between `()` and `anyhow::Error`)

Reply:
Right! My thinking is that if something is wrong, something is wrong. I'll probably not report a profile, will write out an error message, and stop the profiler.

But indeed if you were going for "let's try to keep going", this approach is slightly more annoying (although we could keep a copy of the values -- probably cheap enough with the amount we use).